### PR TITLE
Limit output size

### PR DESCRIPTION
**Pull request by boulter**
Adds a configuration option to limit the maximum size of an output node.

@jmagnarelli-hs

### CODE REVIEW

File: src/main/java/com/hubspot/jinjava/tree/output/OutputList.java [hunk 1]
```diff
@@ -3,12 +3,25 @@
...
+import com.hubspot.jinjava.interpret.OutputTooBigException;
+
 public class OutputList {
 
   private final List<OutputNode> nodes = new LinkedList<>();
   private final List<BlockPlaceholderOutputNode> blocks = new LinkedList<>();
+  private long maxOutputSize;
+
+  public OutputList(long maxOutputSize) {
+
```
Comment (jmagnarelli-hs) [inline src/main/java/com/hubspot/jinjava/tree/output/OutputList.java]: Nit: why the extra line here?

File: src/main/java/com/hubspot/jinjava/tree/output/OutputList.java [hunk 2]
```diff
@@ -3,12 +3,25 @@
...
+    this.maxOutputSize = maxOutputSize;
+  }
 
   public void addNode(OutputNode node) {
+
+    if (maxOutputSize > 0 && node.getSize() > maxOutputSize) {
```
Comment (jmagnarelli-hs) [inline src/main/java/com/hubspot/jinjava/tree/output/OutputList.java]: Does this only check the size of this node against the max? What if we're adding a large number of very small nodes? Just pushed a test that shows what I mean.

If this is only intended to limit the size of individual nodes, I think we should change the name of `maxOutputSize` to something else.

Comment (boulter) [inline src/main/java/com/hubspot/jinjava/tree/output/OutputList.java]: should be fixed now

**Comment by jmagnarelli-hs** [comment]
🍹

**Pull request merged**